src/views/virtualmachines/details/tabs/configuration/scripts/hooks/useDynamicSSHInjectionEnabledForVM.ts

```diff
@@ -0,0 +1,17 @@
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { DYNAMIC_CREDENTIALS_SUPPORT } from '@kubevirt-utils/components/DynamicSSHKeyInjection/constants/constants';
+import useBootableVolumes from '@kubevirt-utils/resources/bootableresources/hooks/useBootableVolumes';
+import { getLabel, getName } from '@kubevirt-utils/resources/shared';
+import { getAccessCredentials } from '@kubevirt-utils/resources/vm';
+import { getPVCSourceOrSourceRef } from '@kubevirt-utils/resources/vm/utils/source';
+
+export const useDynamicSSHInjectionEnabledForVM = (vm: V1VirtualMachine) => {
+  const { name, namespace } = getPVCSourceOrSourceRef(vm);
+  const { bootableVolumes } = useBootableVolumes(namespace);
+  const bootableVolume = bootableVolumes?.find((bv) => getName(bv) === name);
+  const isDynamicSSHInjectionEnabled =
+    getLabel(bootableVolume, DYNAMIC_CREDENTIALS_SUPPORT) === 'true' &&
+    Boolean(getAccessCredentials(vm)?.[0]?.sshPublicKey?.propagationMethod?.qemuGuestAgent);
```

Comment:
this code is very similar to getIsDynamicSSHInjectionEnabled, can you make getIsDynamicSSHInjectionEnabled to get any kind of K8s resource and not just vm?